File: app/src/main/java/com/wapchief/imitationlol/FragmentAdapter.java
package com.wapchief.imitationlol;

import android.content.Context;
import android.support.design.internal.NavigationMenuPresenter;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by apple on 2017/6/21.
 */

public class FragmentAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    List<MainBean> data=new ArrayList<MainBean>();

    public FragmentAdapter(List<MainBean> data, Context context) {
        this.data = data;
        this.context = context;
    }

    private Context context;
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_fragment, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof ItemViewHolder) {
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    int position = holder.getLayoutPosition();
//                    onItemClickListener.onItemClick(holder.itemView, position);
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    private class ItemViewHolder extends RecyclerView.ViewHolder{
        ImageView iv;
        TextView tvTitle,tvContent,tvTime;
        public ItemViewHolder(View view) {
            super(view);
            iv = (ImageView) view.findViewById(R.id.item_img);
            tvTitle = (TextView) view.findViewById(R.id.item_title);
            tvContent = (TextView) view.findViewById(R.id.item_content);
            tvTime = (TextView) view.findViewById(R.id.item_time);
        }
    }
    public interface OnItemClickListener {
        void onItemClick(View view, int position);

        void onItemLongClick(View view, int position);
    }

    private OnItemClickListener onItemClickListener;

    public void setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }
}

File: app/src/main/java/com/wapchief/imitationlol/FragmentAdapter.kt
package com.wapchief.imitationlol

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.*

/**
 * Created by apple on 2017/6/21.
 */
class FragmentAdapter(data: MutableList<MainBean?>, context: Context) : RecyclerView.Adapter<RecyclerView
.ViewHolder>() {
    var datas: MutableList<MainBean?> = ArrayList()
    private val context: Context
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_fragment, parent, false)
        return ItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as? ItemViewHolder)?.itemView?.setOnClickListener {
            val position = holder.getLayoutPosition()
            //                    onItemClickListener.onItemClick(holder.itemView, position);
        }
    }

    override fun getItemCount(): Int {
        return datas.size
    }

    private inner class ItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var iv: ImageView
        var tvTitle: TextView
        var tvContent: TextView
        var tvTime: TextView

        init {
            iv = view.findViewById<View>(R.id.item_img) as ImageView
            tvTitle = view.findViewById<View>(R.id.item_title) as TextView
            tvContent = view.findViewById<View>(R.id.item_content) as TextView
            tvTime = view.findViewById<View>(R.id.item_time) as TextView
        }
    }

    interface OnItemClickListener {
        fun onItemClick(view: View?, position: Int)
        fun onItemLongClick(view: View?, position: Int)
    }

    private var onItemClickListener: OnItemClickListener? = null
    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?) {
        this.onItemClickListener = onItemClickListener
    }

    init {
        this.datas = data
        this.context = context
    }
}